viamonvif/discovery.go

```diff
@@ -305,48 +349,42 @@ func getRTSPStreamURLs(deviceInstance OnvifDevice, username, password string, lo
 	var rtspUris []string
 	// Iterate over all profiles and get the RTSP stream URI for each one
 	for _, profile := range envelope.Body.GetProfilesResponse.Profiles {
-		logger.Debugf("Using profile token: %s", profile.Token)
+		logger.Debugf("Using profile token: %s %#v", profile.Token, profile)
 
 		getStreamURI := media.GetStreamUri{
 			StreamSetup: onvifxsd.StreamSetup{
-				Stream: onvifxsd.StreamType(streamTypeRTPUnicast),
+				Stream:    onvifxsd.StreamType(streamTypeRTPUnicast),
+				Transport: onvifxsd.Transport{Protocol: "rtsp"},
```

Comment:
Was adding transport protocol necessary to discovering that cam you had trouble discovering?

Reply:
yes